packages/trader/src/Stores/Modules/Trading/Helpers/contract-type.ts

```diff
@@ -200,7 +202,9 @@ export const ContractType = (() => {
             case 'Call':
             case 'Put':
                 stored_barriers_data =
-                    v2_params_initial_values?.strike && isDTraderV2()
+                    v2_params_initial_values?.strike &&
+                    (isDtraderV2MobileEnabled(root_store?.ui.is_mobile) ||
```

Comment:
There are a lot of places, where you add check `(isDtraderV2MobileEnabled(root_store?.ui.is_mobile) || isDtraderV2DesktopEnabled(root_store?.ui.is_desktop))`.
Maybe it will make sense to unite it? To create a function, that will check both conditions?

Reply:
const is_dtrader_v2_enabled =
            isDtraderV2DesktopEnabled(ui.is_desktop) || isDtraderV2MobileEnabled(ui.is_mobile);
            
 I have done this in replay-chart.tsx but only because i'm 100% sure that this condition wont be changed down the line. In places where im not sure is the case, i chose to keep them separate.

Reply:
I have the same question here as well, would it affect v1 desktop `stored_barriers_data` if v2_desktop is disabled but v2_mobile is enabled.

Reply:
ok i will look into this

Reply:
i think not. Because of root_store.ui.is_mobile and is_desktop. These flags still have to be true even if feature flags are on or off.

So it'll only apply when device width and feature flags are both true.

Example,

1. mobile feature flag is `true`, desktop feature flag is `false` and we are on `desktop screen`. 
-> `isDtraderMobileEnabled returns false` because we are on desktop screen and` isDtraderDesktopEnabled returns false` because feature flag is off

For stored_barriers_data for v2 will only work if device width is true.

@nijil-deriv

Reply:
Ah, got it.